docs/learn/fundamentals/contract-development/storage/state-archival.mdx

```diff
@@ -37,19 +37,23 @@ The general usage and interface are identical for all storage types. They differ
 ### `Instance`
 
 - Most expensive fees (same price as `Persistent` storage).
-- Archived when TTL goes to 0, can be restored using the [`RestoreFootprintOp`](#restorefootprintop) operation.
+- Archived when TTL goes to 0, is automatically restored via the `InvokeHostFunction` operation.
 - Shares the same TTL as the contract instance. If the contract instance has not been archived, instance data is guaranteed to be accessible and not archived.
 - Limited amount of storage available.
 - Suitable for "shared" contract state that cannot be `Temporary` (i.e. admin accounts, contract metadata, etc.).
 
 ### `Persistent`
 
 - Most expensive fees (same price as `Instance` storage).
-- Archived when TTL goes to 0, can be restored using the [`RestoreFootprintOp`](#restorefootprintop) operation.
+- Archived when TTL goes to 0, is automatically restored via the `InvokeHostFunction` operation.
 - Does not share the same TTL as the contract instance. If the contract instance is not archived, `Persistent` data may be archived and need to be restored before invoking the contract.
 - Unlimited amount of storage.
 - Suitable for user data that cannot be `Temporary` (i.e. balances).
 
+## Contract Data Automatic Restoration
+
+Starting in Protocol 23 ([CAP-0066: Soroban In-Memory Read Resource](https://github.com/stellar/stellar-protocol/blob/master/core/cap-0066.md)), archived `Persistent` or `Instance` contract entries are automatically restored whenever they appear in a footprint during `InvokeHostFunctionOp`. Developers no longer need to issue a separate restore operation -- restoration happens before the host function runs, with the usual fees and rent still applying.
```

Comment:
We can use a more precise link here: https://github.com/stellar/stellar-protocol/blob/master/core/cap-0066.md#automatic-entry-restoration